File: JUNHEE/프로그래머스_week12/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

string solution(string s) {
    string answer = "";
    bool flag = true;
    for(int i = 0; i < s.size(); i++){
        char temp = s[i];
        if(s[i] == ' ') flag = true;
        if(flag && (s[i] >= 'a' && s[i] <= 'z')){
            temp -= 32;
            flag = false;
        }
        else if(flag && s[i] != ' '){
            flag = false;   
        }
        else if(!flag && (s[i] >= 'A' && s[i] <= 'Z')){
            temp += 32;
        }
        answer += temp;
        cout << temp;
    }
    return answer;
}

File: DONGUK/프로그래머스_WEEK_12/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    s = s.lower()
    
    tmp = ''
    for i in range(len(s)):
        if s[i] == ' ':
            answer += tmp
            answer += ' '
            tmp = ''
            
        else:
            if not tmp and 'a' <= s[i] <= 'z':
                tmp += s[i].upper()
            else:
                tmp += s[i]
    
        
        if i == len(s)-1:
            answer += tmp
    return answer